examples/gateway-config/comprehensive.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright Envoy AI Gateway Authors
+# SPDX-License-Identifier: Apache-2.0
+# The full text of the Apache license is available in the LICENSE file at
+# the root of the repo.
+
+apiVersion: aigateway.envoyproxy.io/v1alpha1
+kind: GatewayConfig
+metadata:
+  name: comprehensive-config
+  namespace: default
+spec:
+  extProc:
+    env:
+      - name: OTEL_EXPORTER_OTLP_ENDPOINT
+        value: "http://otel-collector.monitoring:4317"
+      - name: OTEL_EXPORTER_OTLP_HEADERS
+        value: "api-key=your-secret-key"
+      - name: OTEL_SERVICE_NAME
+        value: "ai-gateway-production"
+      - name: OTEL_TRACES_SAMPLER
+        value: "parentbased_traceidratio"
+      - name: OTEL_TRACES_SAMPLER_ARG
+        value: "0.1"
+      - name: LOG_LEVEL
+        value: "info"
+      - name: ENABLE_DEBUG_METRICS
+        value: "false"
+
+    resources:
+      requests:
+        cpu: "100m"
+        memory: "128Mi"
+      limits:
+        cpu: "500m"
+        memory: "512Mi"
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: production-ai-gateway
+  namespace: default
+  annotations:
+    aigateway.envoyproxy.io/gateway-config: comprehensive-config
+spec:
+  gatewayClassName: envoy-gateway
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 8080
+
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: staging-ai-gateway
+  namespace: default
+  annotations:
+    aigateway.envoyproxy.io/gateway-config: comprehensive-config
+spec:
+  gatewayClassName: envoy-gateway
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 8081
```

Comment:
Thanks for the example!

We aim for having always working examples, and this is one of the reasons we use the examples in our `e2e` tests. Can you add a test to our [e2e otel tracing](https://github.com/envoyproxy/ai-gateway/blob/main/tests/e2e/otel_tracing_test.go) that uses this example `GatewayConfig` ?